From d852cc98e6a930886f649b1bac2d2ac54d210f27 Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
Date: Wed, 25 Aug 2021 19:14:01 +0530
Subject: [PATCH 4/9] Add default DS list & few other changes to cloud.cfg

Add mount_default_fields to avoid contention with systemd default
services at random intervals.

Comments in the file describe why they are needed

Signed-off-by: Shreenidhi Shedi <shreenidhi.shedi@broadcom.com>
---
 config/cloud.cfg.tmpl | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/config/cloud.cfg.tmpl b/config/cloud.cfg.tmpl
index 82a7e0fe5..7942bc02b 100644
--- a/config/cloud.cfg.tmpl
+++ b/config/cloud.cfg.tmpl
@@ -87,12 +87,21 @@ apt:
 
 # If you use datasource_list array, keep array items in a single line.
 # If you use multi line array, ds-identify script won't read array items.
+{% if variant in ["photon"] %}
+datasource_list: ['NoCloud', 'ConfigDrive', 'OpenStack', 'VMware', None]
+{% endif -%}
+
 # Example datasource config
 # datasource:
 #   Ec2:
 #     metadata_urls: [ 'blah.com' ]
 #     timeout: 5 # (defaults to 50 seconds)
 #     max_wait: 10 # (defaults to 120 seconds)
+# Set allow_raw_data to true if you want VMware to support raw data feature
+# this will wait 15 sec for cust.cfg
+datasource:
+  VMware:
+    allow_raw_data: false
 
 {% if variant == "amazon" %}
 # Amazon Linux relies on ec2-net-utils for network configuration
-- 
2.48.1

